src/NuGet.Clients/NuGet.Credentials/CredentialService.cs

```diff
@@ -34,7 +34,9 @@ public class CredentialService : ICredentialService
 
         private Action<string> ErrorDelegate { get; }
 
-        public bool HandlesDefaultCredentials { get; }
+        public async Task<bool> HandlesDefaultCredentialsAsync() {
+            return (await _providers).Any(provider => provider is DefaultCredentialsCredentialProvider);
```

Comment:
also, `(await _providers)` assign it in a separate local variable, and then do further processing which will be easy to understand and debug.